docs/source/execute_api.rst

```diff
@@ -0,0 +1,145 @@
+
+Hopefully, we will not get any errors during the notebook execution
+(see the last section for error handling). Note that ``path`` specifies
+in which folder to execute the notebook.
+Finally, to save the resulting notebook:
+
+.. code-block:: python
+
+    nbformat.write(nb, open('executed_notebook.ipynb', mode='wt'))
+
+That's all. Your executed notebook will be saved in the current folder
+in the file *executed_notebook.ipynb*.
+
+Execution arguments
+-------------------
+
+The arguments passed to ``ExecutePreprocessor`` are configuration options
+called `traitlets <http://traitlets.readthedocs.org/>`_.
+There are many cool things about traitlets, for example
+they enforce the type of the input and they can be accessed/modified as
+class attributes. Moreover, each traitlet is automatically exposed
+as command-line options. For example, we can pass the timeout from the
+command-line like this::
+
+    jupyter nbconvert --ExecutePreprocessor.timeout=3600 --to notebook --execute mynotebook.ipynb
+
+Let's now discuss in more detail the two traitlets we used.
+
+The ``timeout`` traitlet defines the maximum time (in seconds) the notebook is
+allowed to run, if the execution takes longer an exception will be raised.
+The default is only 30 s, so in many cases you may want to specify
+an higher value.
+
+The second traitlet, ``kernel_name``, allows specifying the name of the kernel
+to be used for the execution. By default, the kernel name is obtained from the
+notebook metadata. The traitlet ``kernel_name`` allows to specify a user-defined
+kernel, overriding the value in the notebook metadata. A common use case
+is that of a python 2/3 library which includes documentation/testing
+notebooks. These notebooks will specify either a python2 or python3 kernel
+in their metadata
+(depending on the kernel used the last time the notebook was saved).
+In reality, these notebooks will work on both python 2/3 and, for testing,
+it is important to be able to execute them programmatically on both
+versions. Here the traitlet ``kernel_name`` comes to help:
+we can just run each notebook twice, specifying first "python2" and then
+"python3" as kernel name.
+
+Error Handling
+--------------
+
+In the previous sections we saw how to save an executed notebook, assuming
+there are no execution error. But, what if there are errors?
+
+An error during the notebook execution, by default, will stop the execution
+and raise a ``CellExecutionError``. Conveniently, the source cell causing
+the error and the original error name and message are also printed.
+After an error, we can still save the notebook as before:
+
+.. code-block:: python
+
+    nbformat.write(nb, open('executed_notebook.ipynb', mode='wt'))
+
+The saved notebook contains the output up until the failing cell,
+and includes a full stack-trace and error (which can help debugging).
+A pattern I use to execute notebooks while handling errors is the following:
```

Comment:
I think we should reword this to avoid using "I" - it's fine for blog posts, but a bit confusing for docs.